### ISSUE TITLE
Generate some data with custom template function

### ISSUE THREAD

**Issue by maximargo** [brianvoe/gofakeit#279]
Hi! When I try to run the following code I get this error: "template: CodeRun:1: function "title" not defined". What I'm missing?

```
package main

import (
	"fmt"
	"strings"
	"text/template"

	"github.com/brianvoe/gofakeit/v6"
)

func main() {
	f := gofakeit.New(11)
	s := `{{printf "%q" . | title}}`

	funcMap := template.FuncMap{
		"title": strings.Title,
	}

	r, err := f.Template(s, &gofakeit.TemplateOptions{Funcs: funcMap, Data: "the go programming language"})
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(r)
}
```

**Comment by brianvoe** [comment]
hmmm interesting. The only thing I can think of is the pipe call.

@Mrpye Any idea on this one?

**Comment by Mrpye** [comment]
You need to change
```
s := `{{printf "%q" . | title}}`
```
to
```
s := `{{printf "%q" (title .Data)}}`
```

I found an issue in the code, with user functions not getting added to the function map.

```
func templateFuncMap(r *rand.Rand, fm *template.FuncMap) *template.FuncMap {
...

        // Fix merge the user function maps with the template function map
	for k, v := range *fm {
		funcMap[k] = v
	}

	return &funcMap
}

```

**Comment by brianvoe** [comment]
Fixed v6.26.4
